packages/office-ui-fabric-react/src/components/Calendar/CalendarDay.tsx

```diff
@@ -204,6 +219,76 @@ export class CalendarDay extends BaseComponent<ICalendarDayProps, ICalendarDaySt
     }
   }
 
+  private findWeekCorners(weeks: any) {
+    let weekCorners: any = {};
+    let numberOfWeeks = weeks.length;
+    let { firstDayOfWeek, navigatedDate, dateRangeType } = this.props;
+
+    // Get all days in current month
+    let dateRange = getDateRangeArray(navigatedDate, dateRangeType, firstDayOfWeek);
+
+    //Check to see if second to the last day (of the first week) is within the current month
+    if (weeks[0][DAYS_IN_WEEK - 2].originalDate.getMonth() !== navigatedDate.getMonth()) {
+      //There is only one highlighted day in first week, add styles to this square
+      let weekIndex = 0;
+      let dayIndex = DAYS_IN_WEEK - 1;
+      weekCorners[weekIndex + '_' + dayIndex] = 'ms-DatePicker-singleTopDate ' + styles.singleTopDate;
+    }
+    else {
+      //Add styles to first and last highlighted squares in the first week
+      let weekIndex = 0;
+      let dayIndex = dateRange[0].getDay();
```

Comment:
This is not going to work with custom first day of the week, please be sure to test this scenario carefully.